sdks/java/core/src/main/java/org/apache/beam/sdk/coders/CoderRegistry.java

```diff
@@ -250,30 +163,30 @@ public void registerCoder(Class<?> clazz, CoderFactory coderFactory) {
    *
    * @throws CannotProvideCoderException if there is no default Coder.
    */
-  public <InputT, OutputT> Coder<OutputT> getDefaultCoder(
+  public <InputT, OutputT> Coder<OutputT> getCoder(
       TypeDescriptor<OutputT> typeDescriptor,
       TypeDescriptor<InputT> inputTypeDescriptor,
       Coder<InputT> inputCoder)
       throws CannotProvideCoderException {
     checkArgument(typeDescriptor != null);
     checkArgument(inputTypeDescriptor != null);
     checkArgument(inputCoder != null);
-    return getDefaultCoder(
+    return createCoderFromTypeDescriptor(
         typeDescriptor, getTypeToCoderBindings(inputTypeDescriptor.getType(), inputCoder));
   }
 
   /**
    * Returns the {@link Coder} to use on elements produced by this function, given the {@link Coder}
    * used for its input elements.
    */
-  public <InputT, OutputT> Coder<OutputT> getDefaultOutputCoder(
+  public <InputT, OutputT> Coder<OutputT> getOutputCoder(
```

Comment:
I don't think people will use these methods because they are too complicated and are really internal details of how we infer coder propagation for a couple of transforms.